test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -1036,5 +1046,47 @@ function (_, $, Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, IDPDiscoveryF
         });
       });
     });
+
+    Expect.describe('Registration Flow', function () {
+      itp('does not show the registration button if features.registration is not set', function () {
+        return setup().then(function (test) {
+          expect(test.form.registrationContainer().length).toBe(0);
+        });
+      });
+      itp('does not show the registration button if features.registration is false', function () {
+        var registration =  {
+        };
+        return setupRegistrationButton(null, registration).then(function (test) {
+          expect(test.form.registrationContainer().length).toBe(0);
+        });
+      });
+      itp('show the registration button if settings.registration.enable is true', function () {
+        var registration =  {
```

Comment:
Is this a valid usecase? Curious if this can happen actually.

Reply:
my understanding is RaaS and IDP discovery are sort of decoupled feature (thanks to Okta is always to be default idp) therefore we allow registration either primary auth or idp auth.